app/views/projects/samples/attachments/_attachment.html.erb

```diff
@@ -2,18 +2,82 @@
   id="<%= dom_id(attachment) %>"
   class="bg-white border-b dark:bg-slate-800 dark:border-slate-700"
 >
-  <th scope="row" class="px-6 py-4"><%= link_to attachment.file.filename,
-    download_namespace_project_sample_attachment_path(
-      sample_id: @sample.id,
-      id: attachment.id
-    ),
-    data: {
-      turbo: false
-    } %></th>
-  <td class="px-6 py-4"><%= number_to_human_size(attachment.file.blob.byte_size) %></td>
+  <% if attachment.associated_attachment && attachment.metadata['direction'] == 'forward' %>
+    <td class="px-6 py-4">
+      <div>
+        <div class="flex mb-4 items-center">
+          <%= viral_icon(name: :arrow_right, color: :subdued, classes: "h-6 w-6 ml-0 mr-2") %>
+          <span>
+            <%= link_to attachment.file.filename,
+            download_namespace_project_sample_attachment_path(
+              sample_id: @sample.id,
+              id: attachment.id
+            ),
+            data: {
+              turbo: false
+            } %>
+          </span>
+        </div>
+        <div class="flex items-center">
+          <%= viral_icon(name: :arrow_left, color: :subdued, classes: "h-6 w-6 ml-0 mr-2") %>
+          <span>
+            <%= link_to attachment.associated_attachment_filename,
+            download_namespace_project_sample_attachment_path(
+              sample_id: @sample.id,
+              id: attachment.metadata["associated_attachment_id"]
+            ),
+            data: {
+              turbo: false
+            } %>
+          </span>
+        </div>
+      </div>
+    </td>
+  <% else %>
+    <td class="px-6 py-4"><%= link_to attachment.file.filename,
+      download_namespace_project_sample_attachment_path(
+        sample_id: @sample.id,
+        id: attachment.id
+      ),
+      data: {
+        turbo: false
+      } %>
+    </td>
+  <% end %>
+
+  <% if attachment.metadata['format'] == 'fasta' %>
```

Comment:
Let's move this `if/elsif/elsif` block into a new helper under `app/helpers` and then call it with <% format_color = helper_method_name(attachment) %>. Reason for this is we will probably need to list the sample files else where and then we have a shared method to get the correct color.

Reply:
I waffled over whether to make the pill color helper extremely generic to be usable by things outside of attachments (c43dd9224b0a9916ecf5fadf7cbc4e2bd466ae58) but decided maybe it's best to have a method within the helper specific to attachments, for less confusion with the arguments required to be passed.

Let me know what you think f93ae17f6874db9352897610bc2fa5ff92ec5f3a

Reply:
Fixed in 9fde28e1bab0cebc9ae7142848f7aa2f747db687